linked-list.rb

```diff
@@ -51,25 +51,52 @@ def delete(val)
    end
 
    def display
-       # Traverse through the list till you hit the "nil" at the end
-       current = @head
-       full_list = []
-       while current.next_node != nil
-           full_list += [current.value.to_s]
-           current = current.next_node
-       end
-       full_list += [current.value.to_s]
-       puts full_list.join("->")
-   end
-
-   def include?(key)
+     # Traverse through the list till you hit the "nil" at the end
+     current = @head
+     full_list = []
+     while current.next_node != nil
+        full_list += [current.value.to_s]
+        current = current.next_node
+     end
+     full_list += [current.value.to_s]
+     puts full_list.join("->")
    end
 
-   def size
-   end
-
-   def max
-   end
+  def include?(key)
+    current = @head
+    # breaks out when we reach last instead of check last
+    while current.next_node != nil
+      return true if current.value == key
+      current = current.next_node
+    end
+    return true if current.value == key
+    return false
+  end
+
+  def size
+    current = @head
+    size = 1
+    while current.next_node != nil
+      size += 1
+      current = current.next_node
+    end
+    size
```

Comment:
I know ruby says it's ok not to put the return in front of this, but I think it's weird :)